src/app/services/Interceptor/MyHttpInterceptor.ts

```diff
@@ -1,18 +1,48 @@
+// my-http-interceptor.ts
 import { Injectable } from '@angular/core';
 import { HttpInterceptor, HttpHandler, HttpRequest, HttpEvent } from '@angular/common/http';
 import { Observable } from 'rxjs';
+import { finalize, delay } from 'rxjs/operators';
 
 @Injectable()
 export class MyHttpInterceptor implements HttpInterceptor {
+
+  private activeRequests = 0;
+  private loadingElement: HTMLElement | null = null;
+  private readonly MIN_DISPLAY_TIME = 500; // Minimum display time in milliseconds
+
+  private showLoader(): void {
```

Comment:
Loader logic code should be in a different file like utils folder and you need to import here.